qutebrowser/misc/sessions.py

```diff
@@ -204,6 +205,10 @@ def _save_tab(self, tab, active):
         data = {'history': []}
         if active:
             data['active'] = True
+
+        if config.val.tabs.mode_on_change == 'restore':
+            data['input_mode'] = tab.data.input_mode.value
```

Comment:
It'd be much nicer to save `.name` (a string) instead of `.value` (an int) here. You can use `usertypes.KeyMode[...]` to get the enum object back.

Reply:
Using `tab.data.input_mode` here means the current tab's mode won't be saved. It probably should be, or `tab.data.input_mode` should be written on any mode change.